netmiko/garderos/garderos_ssh.py

```diff
@@ -0,0 +1,488 @@
+        # Verify success
+        if not save_config_result.__contains__(
+            "Values are persistently saved to STARTUP-CONF"
+        ):
+            raise ValueError(
+                f"Saving configuration was unsuccessful. Device said: {save_config_result}"
+            )
+        # Return device output
+        save_config_result = str(save_config_result)
+        return save_config_result
+
+    def check_linux_mode(self, check_string: str = "]#", pattern: str = "#") -> bool:
+        """Checks if the device is in Linux mode or not.
+
+        :param check_string: Identification of configuration mode from the device
+
+        :param pattern: Pattern to terminate reading of channel
+        """
+        self.write_channel(self.RETURN)
+        output = self.read_until_prompt(read_entire_line=True)
+        return check_string in output
+
+    def linux_mode(self, linux_command: str = "linux-shell", pattern: str = "") -> str:
+        """Enter into Linux mode.
+
+        :param config_command: Linux command to send to the device
+
+        :param pattern: Pattern to terminate reading of channel
+        """
+        output = ""
+        if not self.check_linux_mode():
+            self.write_channel(self.normalize_cmd(linux_command))
+            output = self.read_until_pattern(pattern=pattern)
+            if not self.check_linux_mode():
+                raise ValueError("Failed to enter Linux mode.")
+        return output
+
+    def exit_linux_mode(self, exit_linux: str = "exit", pattern: str = "#") -> str:
+        """Exit from Linux mode.
+
+        :param exit_config: Command to exit Linux mode
+
+        :param pattern: Pattern to terminate reading of channel
+        """
+        output = ""
+        if self.check_linux_mode():
+            self.write_channel(self.normalize_cmd(exit_linux))
+            output = self.read_until_pattern(pattern=pattern)
+            if self.check_linux_mode():
+                raise ValueError("Failed to exit Linux mode")
+        return output
+
+    def remove_and_replace_control_chars(self, s: str) -> str:
+        """Removing all occurrences of "\r\n\r" except of the last occurrence
+        Last occurence will be replaced by "\n"
+
+        :param s: String that needs to be cleansed
+        """
+        # Because the sequence "\r\n\r" also matches "\r\n\r\n",
+        # we need to replace "\r\n\r\n" with "\n\n" first
+        s = s.replace("\r\n\r\n", "\n\n")
+        # Now we have eliminated "\r\n\r\n"
+        # and can begin working on the remaining "\r\n\r" occurrences
+        control_seq = "\r\n\r"
+        if s.count(control_seq) == 0:
+            return s
+        else:
+            index_last_occurrence = s.rfind(control_seq)
+            index_rest_of_string = index_last_occurrence + len(control_seq)
+            return (
+                s[:index_last_occurrence].replace(control_seq, "")
+                + "\n"
+                + s[index_rest_of_string:]
+            )
```

Comment:
Do you have examples of what the output looks like before any modification? It seems strange to me that you just remove all of the `\r\n\r` sequences except for the last one?

Reply:
First of all, thanks for reviewing my code.
Back in November, I definitely had several routers that needed this fix - even if this looks strange. Unfortunately, I couldn't reproduce it today.
I suggest that I remove this part for now and submit a new PR. I might need to add it again later on if I discover this is getting an issue again.

Reply:
Yeah, if you see the output let me know. I somewhat suspect it is a Netmiko artifact (i.e. the newline conversion isn't working a 100% right and leaves some strange <enter> artifacts. I have seen something similar on Cisco NX-OS which I never fully worked out.